jax_galsim/convolve.py

```diff
@@ -0,0 +1,331 @@
+
+        # Figure out what gsparams to use
+        if gsparams is None:
+            # If none is given, take the most restrictive combination from the obj_list.
+            self._gsparams = GSParams.combine([obj.gsparams for obj in args])
+        else:
+            # If something explicitly given, then use that.
+            self._gsparams = GSParams.check(gsparams)
+
+        # Apply gsparams to all in obj_list.
+        if self._propagate_gsparams:
+            self._obj_list = [obj.withGSParams(self._gsparams) for obj in args]
+        else:
+            self._obj_list = args
+
+        # Save the list of objects
+        self._params = {"obj_list": self._obj_list}
+
+    @property
+    def obj_list(self):
+        """The list of objects being convolved."""
+        return self._obj_list
+
+    @property
+    def real_space(self):
+        """Whether this `Convolution` should be drawn using real-space convolution rather
+        than FFT convolution.
+        """
+        return self._real_space
+
+    def withGSParams(self, gsparams=None, **kwargs):
+        """Create a version of the current object with the given gsparams
+
+        .. note::
+
+            Unless you set ``propagate_gsparams=False``, this method will also update the gsparams
+            of each object being convolved.
+        """
+        if gsparams == self.gsparams:
+            return self
+        from copy import copy
+
+        ret = copy(self)
+        ret._gsparams = GSParams.check(gsparams, self.gsparams, **kwargs)
+        if self._propagate_gsparams:
+            ret._obj_list = [obj.withGSParams(ret._gsparams) for obj in self.obj_list]
+        return ret
+
+    def __eq__(self, other):
+        return self is other or (
+            isinstance(other, Convolution)
+            and self.obj_list == other.obj_list
+            and self.real_space == other.real_space
+            and self.gsparams == other.gsparams
+            and self._propagate_gsparams == other._propagate_gsparams
+        )
+
+    def __hash__(self):
+        return hash(
+            (
+                "galsim.Convolution",
+                tuple(self.obj_list),
+                self.real_space,
+                self.gsparams,
+                self._propagate_gsparams,
+            )
+        )
+
+    def __repr__(self):
+        return (
+            "galsim.Convolution(%r, real_space=%r, gsparams=%r, propagate_gsparams=%r)"
+            % (
+                self.obj_list,
+                self.real_space,
+                self.gsparams,
+                self._propagate_gsparams,
+            )
+        )
+
+    def __str__(self):
+        str_list = [str(obj) for obj in self.obj_list]
+        s = "galsim.Convolve(%s" % (", ".join(str_list))
+        if self.real_space:
+            s += ", real_space=True"
+        s += ")"
+        return s
+
+    # JEC not sure if it is used
+    # def _prepareDraw(self):
+    #    for obj in self.obj_list:
+    #        obj._prepareDraw()
```

Comment:
remove this commented out code?

Reply:
sure... not sure what prepareDraw was supposed to do

Reply:
I removed it